File: validanagram2.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
        {
            return false;
        }

        map<char, int> m1;
        map<char, int> m2;

        for (char c : s)
        {
            m1[c]++;
        }

        for (char c : t)
        {
            m2[c]++;
        }

        if (m1 == m2)
        {
            return true;
        }
        return false;
    }
};

File: validanagram2.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        TreeMap<Character, Integer> map = new TreeMap<>();
        TreeMap<Character, Integer> map2 = new TreeMap<>();

        for (char c : s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            map2.put(c, map2.getOrDefault(c, 0) + 1);
        }

        return map.equals(map2);
    }
}